content/chainguard/chainguard-enforce/chainguard-enforce-kubernetes/detect-log4shell-demo.md

```diff
@@ -0,0 +1,296 @@
+
+For more information on how policies work in Chainguard Enforce, check out [Sigstore’s documentation on how it handles and reads policies](https://docs.sigstore.dev/policy-controller/overview/).
+
+After creating the policy file, you can use it to activate a policy for your cluster by running the following command.
+
+```sh
+chainctl policies create log4shell-demo-policy \
+    --group $GROUP_ID \
+    -f log4shell-demo-policy.yaml
+```
+
+To confirm that the policy was successfully created, run the following command to retrieve a list of all your active policies.
+
+```sh
+chainctl policies ls
+```
+
+Ensure that your new policy has ended up in the config-image-policies configmap by running this command.
+
+```sh
+kubectl -n cosign-system get configmap config-image-policies -o yaml
+```
+
+If the policy is in the configmap, this command's output will include a "log4 shell-demo-policy" key.
+
+You can find all records in the cluster containing SBOMs indicating the Log4Shell vulnerability using the following command.
+
+```sh
+chainctl cluster records ls $CLUSTER
+```
+```
+                                       IMAGE                                      |            POLICIES            |   WORKLOADS   | PACKAGES | LAST SEEN | LAST REFRESHED  
+----------------------------------------------------------------------------------+--------------------------------+---------------+----------+-----------+-----------------
+  ghcr.io/chainguard-dev/log4shell-demo/app@sha256:ba4037…                        | log4shell-demo-policy:fail:40s | Pod:1         | maven:2  | 3m49s     | sbom:39s        
+. . .
+```
+
+Information about the SBOM and the policy violation will also appear in the UI.
+
+## Step 4 – Cleaning Up Your System
+
+After confirming that the policy you set causes Chainguard Enforce to detect the images vulnerable to the Log4Shell vulnerability, it's important that you clean up your system.
+
+Start by deleting the pod running the vulnerable application.
+
+```sh
+kubectl delete pod log4shell-demo
+```
+
+Then delete the policy. 
+
+```sh
+chainctl policy delete -y log4shell-demo-policy
+```
+
+This will cause the policy to be removed from the policy list in both the CLI and UI.
+
+Lastly, uninstall the agent from your cluster.
+
+```sh
+chainctl cluster uninstall
+```
+
+
+## Learn More
+
+By following this guide, you will have a better understanding of how to set up a Chainguard Enforce policy. You can use this understanding to apply other policies to your cluster. 
+
+To continue your learning, we encourage you to check out this [sample policy for Enforce published by Chainguard](https://github.com/chainguard-dev/text4shell-policy). This policy protects against the [Text4Shell vulnerability](https://nvd.nist.gov/vuln/detail/CVE-2022-42889). When installed into a cluster, this policy will identify and optionally block workloads containing vulnerable versions of Apache Commons Text from running on customer clusters.
```

Comment:
Should we make this example a full guide too?

Reply:
If you agree about a full guide for this, can you open an issue for it?

Reply:
Added! https://github.com/chainguard-dev/edu/issues/184